.github/workflows/publish_images.yaml

```diff
@@ -70,17 +70,29 @@ jobs:
             charmcraft release "${charm_name}" --revision=${charm_rev} --channel=edge --resource="${charm_resource}:${img_rev}"
           }
 
-          for release in $(ls $GITHUB_WORKSPACE/legend/releases/)
-                    
-          do
-            # If there was a new Legend release yesterday, release it to Charmhub
-            if [[ "$release" == $(date +%Y-%m-%d -d "yesterday") ]];          
+          previous_release=""
+          
+          for release in $(ls $GITHUB_WORKSPACE/legend/releases/ | sort -r)
```

Comment:
oh I see, you're reversing them, so you publish the latest release first, so if there are multiple releases in the same month, we only push the latest one. Interesting. A comment explaining this would be great.

NOTE: this is a bit problematic though. If you check above in ``release_charm_with_image``, we're releasing the given charm version as the latest, which means that with these changes, we'd be publishing the oldest charm as the latest, which is not true. We should only do that for the latest release. Btw, this also happens with multiple releases per month. For example, let's consider the following releases:

```
2023-05-24
2023-05-30
```

We're publishing charms per yyyy-mm, not the entire date. Because you're going in reverse order, you'll be publishing the oldest release in the month as the latest in the month, which is not correct.

Also, we used to not treat older releases than 2022.04.01, if you check git blame:

```
          # Treat releases newer than 2022.04.01.
          for release in $(ls $GITHUB_WORKSPACE/legend/releases/ | gawk '{ if ($1 > "2022-04-01") { print } }')
```

We should keep that behaviour.

Reply:
This might accomplish what we need to do:

```
ls -r releases/ | gawk '{ if ($1 > "2022-04-01") { print } }' | uniq -w 7 | sort
```